libqtile/utils.py

```diff
@@ -424,9 +442,37 @@ async def add_signal_receiver(
     )
 
     # Check if message sent successfully
-    if msg and msg.message_type == MessageType.METHOD_RETURN:
+    if bus and msg and msg.message_type == MessageType.METHOD_RETURN:
         bus.add_message_handler(callback)
         return True
 
     else:
         return False
+
+
+async def find_dbus_service(service: str, session_bus: bool) -> bool:
+    """Looks up service name to see if it is currently available on dbus."""
+
+    # We're using low level interface here to reduce unnecessary calls for
+    # introspection etc.
+    bus, msg = await _send_dbus_message(
+        session_bus,
+        MessageType.METHOD_CALL,
+        "org.freedesktop.DBus",
+        "org.freedesktop.DBus",
+        "/org/freedesktop/DBus",
+        "ListNames",
+        "",
+        [],
+    )
+
+    if bus is None or msg is None or (msg and msg.message_type != MessageType.METHOD_RETURN):
+        logger.warning("Unable to send lookup call to dbus.")
+        return False
+
+    bus.disconnect()
+    del bus
```

Comment:
why do we need to del bus?

Reply:
I thought dbus-next had a `__del__` method for disconnecting so had that in first but realised I was wrong and I should really `disconnect()` instead. We can remove the `del` statement.